src/main/java/de/qabel/desktop/daemon/management/AbstractTransaction.java

```diff
@@ -1,13 +1,14 @@
 package de.qabel.desktop.daemon.management;
 
+import java.nio.file.Path;
 import java.util.LinkedList;
 import java.util.List;
 import java.util.Observable;
 import java.util.concurrent.CopyOnWriteArrayList;
 
 import static de.qabel.desktop.daemon.management.Transaction.STATE.*;
 
-public abstract class AbstractTransaction extends Observable implements Transaction {
+public abstract class AbstractTransaction<S extends Path, D extends Path> extends Observable implements Transaction<S,D> {
     public static final long METADATA_SIZE = 56320L;
```

Comment:
look at line 20:
```JAVA
private Long size = METADATA_SIZE;  // metadata size ... imagine a random number here
```
it was just used for the user to show some progress even when only directories are synced ... we need a better approximation in the future however